### ISSUE TITLE
[BUG]: DPad input not detected for XBox wireless controller on linux

### ISSUE THREAD

**Issue by 2000Slash** [PCSX2/pcsx2#8483]
### Describe the Bug

I'm trying to play with an xbox wireless controller over usb (output of lsusb: ID 045e:0b12 Microsoft Corp. Xbox Wireless Controller (model 1914)). I use the flatpak version of pcsx2, since I'm on linux. I played on nightly v1.7.4063 until I updated today to v1.7.4274. Since then, pcsx2 won't recognize the dpad input of my controller anymore. I **can** bind other controller buttons to the dpad input setting, but I **can't** bind the actual dpad on the controller to anything.

### Reproduction Steps

Connect xbox one controller over usb to pc.
Start pcsx2.
Go to Settings -> Controllers -> Controller Port 1 DualShock 2
Click on any input field.
Click any direction on the dpad.
Nothing gets bound

### Expected Behavior

The dpad buttons should be able to be used

### PCSX2 Revision

v1.7.4274

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

Arch

**Comment by Mrlinkwii** [comment]
>I use the flatpak version of pcsx2

flatpak isnt supported , use the  appimage

**Comment by 2000Slash** [comment]
I have the same issue on appimage version v1.7.4274

**Comment by Mrlinkwii** [comment]
dose automapping work?

**Comment by 2000Slash** [comment]
Automapping does assign the mappings correctly (as it was before), but the dpad still does nothing ingame when pressed

**Comment by Mrlinkwii** [comment]
which game may i ask? :)

**Comment by 2000Slash** [comment]
I am using Gran Turismo 4 pal SCES-51791. But I'm also open to test on other things if you want :+1:

**Comment by 2000Slash** [comment]
I researched a bit and found the bug. in SDLInputSource.cpp the HandleJoystickHatEvent method handles the dpad button presses. When debugged here, they come through successfully. I also found: 
```
// Track last hat state so we can send "unpressed" events.
std::vector<u8> last_hat_state;
```
However when I debug in HandleJoystickHatEvent, ev->hat is 0 and it->last_hat_state is empty so the function returns:
```
if (it == m_controllers.end() || ev->hat >= it->last_hat_state.size())
	return false;
```
I still don't know why this works on my windows machine though :thinking:
